astroquery/nasa_ads/core.py

```diff
@@ -102,10 +107,36 @@ def _parse_response(self, response):
 
         return t
 
-    def _args_to_payload(self, query_string):
+    def _args_to_url(self, query_string):
         # convert arguments to a valid requests payload
         # i.e. a dictionary
-        return {'qsearch': query_string, 'data_type': 'XML'}
+        request_string = 'q=' + urlencode(query_string)
+        return request_string
+
+    def _fields_to_url(self):
+        request_fields = '&fl=' + ','.join(self.ADS_FIELDS)
+        return request_fields
+
+    def _get_token(self):
+        """
+        Try to get token from the places Andy Casey's python ADS client expects it, otherwise return an error
+        """
+        if self.TOKEN is not None:
+            return self.TOKEN
+
+        self.TOKEN = os.environ.get('ADS_DEV_KEY', None)
+        if self.TOKEN is not None:
+            return self.TOKEN
+
+        token_file = os.path.expanduser('~/.ads/dev_key')
```

Comment:
Could use os.path.join to make sure it works on windows, too

Reply:
According to the docs, *theoretically* os.path.expanduser should work on either Windows or Unix. Unless you've seen otherwise?

Reply:
Oh, I didn't mean it for the `~` part, but for the separators in `/.ads/dev_key`

Reply:
Ooh right, good call.